File: ffrontend/src/app/commons/dtos/DtoAlgorithmAdd.ts
import { input, output } from "@angular/core";


export interface BAlgorithmAdd{

    jobId : number;
    name : string;
    round : number;
    title : string;
    category : string;
    difficulty : string;
    timeLimit : number;
    description : string;
    inputFormat : string;
    outputFormat : string;
    timeComplexity : string;
    spaceComplexity : string;
    sampleSolution : string;
    constraints : Array<string>;
    examples : Array<BExample>;
    hints : Array<string>;
    testCases : Array<BTestCaseAlgorithm>;
}
export interface BExample{
    input : string;
    output : string;
    explanation : string;
}

export interface BTestCaseAlgorithm{
    input : string;
    output : string;
    hidden : boolean;
    points : number;
}

export interface finishProg{
    kerdoivId : number;
    programkod : string;
}

File: allasinterju-backend/Allasinterju.Core/DTOs/DtoAlgorithmAdd.cs
public class BAlgorithmAdd{
    public int JobId{get;set;}
    public string Name{get;set;}
    public int Round{get;set;}
    public string Title{get;set;}
    public string Category{get;set;}
    public string Difficulty{get;set;}
    public int TimeLimit{get;set;}
    public string Description{get;set;}
    public string InputFormat{get;set;}
    public string OutputFormat{get;set;}
    public string TimeComplexity{get;set;}
    public string SpaceComplexity{get;set;}
    public string SampleSolution{get;set;}
    public List<string> Constraints{get;set;}
    public List<BExample> Examples{get;set;}
    public List<string> Hints{get;set;}
    public List<BTestCaseAlgorithm> TestCases{get;set;}

}
public class BExample{
    public string Input{get;set;}
    public string Output{get;set;}
    public string Explanation{get;set;}
}
public class BTestCaseAlgorithm{
    public string Input{get;set;}
    public string Output{get;set;}
    public bool Hidden{get;set;}
    public int Points{get;set;}
}